### PR TITLE
CSHARP-2482: Fix transaction test runner

### PR DESCRIPTION
**Pull request by vincentkam**
https://evergreen.mongodb.com/version/5ccc85c01e2d1706c1b43c6d

### CODE REVIEW

**Comment by vincentkam** [comment]
Ready for review!

File: tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs [hunk 1]
```diff
@@ -399,7 +399,7 @@ private DisposableBundle ConfigureFailPointOnPrimaryOrShardRoutersIfNeeded(IMong
 
         private void ExecuteOperations(IMongoClient client, Dictionary<string, object> objectMap, BsonDocument test)
         {
-            var factory = new JsonDrivenTestFactory(client, _databaseName, _collectionName, bucketName: null, objectMap);
+            var factory = new JsonDrivenTestFactory(this, client, _databaseName, _collectionName, bucketName: null, objectMap);
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs:402]: This is what happens when you leave in the old constructor for backward compatibility... :)

Was this a merge conflict resolution problem? Looks like it was.

**Review by rstam — APPROVED** [review]

**Comment by vincentkam** [comment]
Pushed! 😄

File: tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs (hunk 1 above)
Comment (vincentkam) [reply tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs:402]: Agreed, it probably would have been better to update the old constructor! 

Also agreed: I think it was probably a merge conflict resolution problem

**Pull request merged**